arcgis-ios-sdk-samples/Utility Network/Configure subnetwork trace/ConfigureSubnetworkTraceViewController.swift

```diff
@@ -0,0 +1,391 @@
+            addConditionLabel.isEnabled = true
+            resetLabel.isEnabled = true
+            traceLabel.isEnabled = true
+        }
+    }
+    
+    // Convert the expression into a readable string.
+    func expressionToString(expression: AGSUtilityTraceConditionalExpression) -> String? {
+        if let categoryComparison = expression as? AGSUtilityCategoryComparison {
+            let comparisonOperatorString = categoryComparisonOperators[categoryComparison.comparisonOperator]
+            return "`\(categoryComparison.category.name)` \(comparisonOperatorString!)"
+        } else if let attributeComparison = expression as? AGSUtilityNetworkAttributeComparison {
+            // Check if attribute domain is a coded value domain.
+            if let domain = attributeComparison.networkAttribute.domain as? AGSCodedValueDomain {
+                // Get the coded value using the the attribute comparison value and attribute data type.
+                let dataType = attributeComparison.networkAttribute.dataType
+                let attributeValue = convertToDataType(otherValue: attributeComparison.value!, dataType: attributeComparison.networkAttribute.dataType)
+                let codedValue = domain.codedValues.first(where: { compare(dataType: dataType, comparee1: $0.code!, comparee2: attributeValue!) })
+                let comparisonOperatorString = comparisonsStrings[attributeComparison.comparisonOperator.rawValue]
+                return "'\(attributeComparison.networkAttribute.name)' \(comparisonOperatorString) '\(codedValue!.name)'"
+            } else {
+                if let nameOrValue = attributeComparison.otherNetworkAttribute?.name {
+                    let comparisonOperatorString = comparisonsStrings[attributeComparison.comparisonOperator.rawValue]
+                    return "`\(attributeComparison.networkAttribute.name)` \(comparisonOperatorString) `\(nameOrValue)`"
+                } else if let nameOrValue = attributeComparison.value {
+                    let comparisonOperatorString = comparisonsStrings[attributeComparison.comparisonOperator.rawValue]
+                    return "`\(attributeComparison.networkAttribute.name)` \(comparisonOperatorString) `\(nameOrValue)`"
+                }
+            }
+        } else if let andCondition = expression as? AGSUtilityTraceAndCondition {
+            return "(\(expressionToString(expression: andCondition.leftExpression)!)) AND\n(\(expressionToString(expression: andCondition.rightExpression)!))"
+        } else if let orCondition = expression as? AGSUtilityTraceOrCondition {
+            return "(\(expressionToString(expression: orCondition.leftExpression)!)) AND\n(\(expressionToString(expression: orCondition.rightExpression)!))"
+        }
+        return nil
+    }
+    
+    // Convert the values to matching data types.
+    func convertToDataType(otherValue: Any, dataType: AGSUtilityNetworkAttributeDataType) -> Any? {
```

Comment:
It is tricky to convert string values into numbers. I tried with an evil 👿  case to input `010101010101010101010` to the value's row, and the force unwrap will crash. Guess that is overflowed.